flink-end-to-end-tests/test-scripts/test_resume_externalized_checkpoints.sh

```diff
@@ -125,9 +126,20 @@ fi
 
 echo "Restoring job with externalized checkpoint at $CHECKPOINT_PATH ..."
 
-BASE_JOB_CMD=`buildBaseJobCmd $NEW_DOP`
+BASE_JOB_CMD=`buildBaseJobCmd $NEW_DOP "-s file://${CHECKPOINT_PATH}"`
+JOB_CMD=""
+if [[ $SIMULATE_FAILURE == "true" ]]; then
```

Comment:
Why do we construct a failing job for the recovery step? Wouldn't a simple job which tests the resuming without explicit failure be sufficient?

Reply:
Thanks for looking into it @tillrohrmann!
We construct a failing job (that doesn't actually fail since all the parameters are 0) so that the recovered job and the original job would have exactly the same operators (omitting the `--test.simulate_failure true` omits the `FailureMapper` from the recovered job.)
Having the jobs differ doesn't allow recovering from the externalized checkpoint without supplying `--allowNonRestoredState` (unless there is a different way that I'm not aware of)
The problem with passing this argument is:
1. I think It makes the test more fragile , if the recovered job would accidentally start with a fresh state. 
2. The `SemanticsCheckMapper` reports violation in the following case:
```run_test "Running HA per-job cluster (file, sync) end-to-end test" "$END_TO_END_DIR/test-scripts/test_ha_per_job_cluster_datastream.sh file false false" "skip_check_exceptions"
``` which looks like an artifact of how a job is constructed from its arguments via `DataStreamAllroundTestJobFactory` and the missing `.uid()` but we can look into that in a followup ticket?